app/controllers/works_controller.rb

```diff
@@ -1,11 +1,13 @@
 # FRONT-END controller for public Works views -- mostly just the #show action.
 # Not to be confused with admin/works_controller for staff/management/edit views.
 class WorksController < ApplicationController
-  before_action :set_work, :check_auth
+  before_action :set_work, :set_decorator_and_template, :check_auth
 
```

Comment:
rather than a before action, if you are using a different decorator for the different cases anyway, make this `decorator.view_template`. 

If you were not using decorators, I would still make this a `view_template_name` method in in controller, that returns the right one. 

I know Rails uses instance variables sometimes, but too many of them get confusing, where was the state set by whom what is it now where can it be changed? A method with `||=`-style memoization is better.

Reply:
OK. It still feels kinda wrong this way. I'll try `decorator.view_template`; that sounds less ugly.